### PR TITLE
[ci skip] `define_model_callbacks` only exist in active model

### PR DESCRIPTION
**Pull request by Edouard-chin**
Small documentation fix

### CODE REVIEW

**Comment by Edouard-chin** [comment]
I'm wondering if it would be worth mentioning in the docs that calling `define_callbacks` twice will actually remove the callbacks already set as it's creating a new instance of `CallbackChain`

```ruby
define_callbacks :validate

set_callback :validate, ->(o) { o.returns_true }
define_callbacks :validate # This removes the previous callback we set
```

**Comment by maclover7** [comment]
^ Do you want to submit that as a separate PR for that?

**Comment by Edouard-chin** [comment]
If you think it's valuable then I can add it in this PR. I ran into the problem, having a line in the doc mentioning it would have probably saved me a headache 😅

File: activesupport/lib/active_support/callbacks.rb [hunk 1]
```diff
@@ -736,8 +736,13 @@ def reset_callbacks(name)
...
       #   would call <tt>Audit#save</tt>.
       #
-      # NOTE: +method_name+ passed to `define_model_callbacks` must not end with
+      # ===== Notes
+      #
+      # +names+ passed to `define_callbacks` must not end with
       # `!`, `?` or `=`.
+      #
+      # Calling `define_callbacks` multiple times with the same +names+ will
+      # overwrite previous callback set with `set_callback`.
```
Comment (maclover7) [inline activesupport/lib/active_support/callbacks.rb]: previous callbacks

**Comment by maclover7** [comment]
Seems reasonable to me. Added a comment with one or two small grammar changes :)

**Comment by Edouard-chin** [comment]
thanks for your help 😸

File: activesupport/lib/active_support/callbacks.rb [hunk 2]
```diff
@@ -736,8 +736,13 @@ def reset_callbacks(name)
       #
       #   would call <tt>Audit#save</tt>.
       #
-      # NOTE: +method_name+ passed to `define_model_callbacks` must not end with
+      # ===== Notes
```
Comment (vipulnsward) [inline activesupport/lib/active_support/callbacks.rb]: Did you follow this from somewhere?

Comment (Edouard-chin) [inline activesupport/lib/active_support/callbacks.rb]: yes, https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/http_authentication.rb#L161

I just added 2 more `=` to be consistent with the `===== Option` [above](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/callbacks.rb#L599)

**Comment by vipulnsward** [comment]
Thanks @Edouard-chin !

**Pull request merged**